### PR TITLE
[FEATURE] Pre and post import eventDispatchers

### LINKED ISSUE #1849: Pre and post import eventDispatchers
How to keep news items in sync with an external source?

For example by deleting all previous records of a certain pid.

In the NewsImportService class there are 2 eventDispatchers which fire before processing a record and after. But there is no way to do custom stuf before or after the whole import process.

So, my question is can I propose code to add 2 additional eventDispatchers in the import function?

### PR DESCRIPTION
**Pull request by Zillion01**
Resolves #1849

### CODE REVIEW

File: Classes/Event/NewsPostImportEvent.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?php
+
```
Comment (georgringer) [inline Classes/Event/NewsPostImportEvent.php:2]: please add `declare(strict_types=1);`

File: Classes/Event/NewsPreImportEvent.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?php
+
```
Comment (georgringer) [inline Classes/Event/NewsPreImportEvent.php:2]: please add `declare(strict_types=1);`

File: Classes/Event/NewsPostImportEvent.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    private $importData;
+
+    public function __construct(NewsImportService $newsImportService, array $importData)
+    {
+        $this->newsImportService = $newsImportService;
+        $this->importData = $importData;
+    }
+
+    /**
```
Comment (georgringer) [inline Classes/Event/NewsPostImportEvent.php:31]: all those comments can be removed, also in other event

**Review by georgringer — CHANGES_REQUESTED** [review]

**Comment by Zillion01** [comment]
Thanks a lot

**Comment by georgringer** [comment]
thanks to you ;)

**Pull request merged**